packages/shared/src/store/sync.ts

```diff
@@ -102,10 +103,23 @@ export const handleChannelsUpdate = async (update: api.ChannelsUpdate) => {
             replyTime: update.post.sentAt,
           });
         }
+        // insert the reply
+        await db.insertChannelPosts({
+          channelId: update.post.channelId,
+          posts: [update.post],
+        });
       }
-
-      // finally, always insert the post itself
-      await db.insertChannelPosts(update.post.channelId, [update.post]);
+      // For non-replies, we skip inserting and sync from the api instead.
+      // This is significantly slower, but ensures that windowing stays correct. Better solutions would be:
+      // 1. Have the server send the previous post id so that we can match this up with an older window if it exists
```

Comment:
This would be easy to add to the backend

Reply:
Ya, @Fang- and I discussed some better ideas here -- will update with the changes endpoint.